sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala

```diff
@@ -324,4 +324,16 @@ class QueryExecutionErrorsSuite extends QueryTest
     assert(e5.getSqlState === "42000")
     assert(e5.getMessage === "Cannot parse decimal")
   }
+
+  test("CAST_CAUSES_OVERFLOW") {
+    val e = intercept[SparkArithmeticException] {
+      sql("set spark.sql.ansi.enabled=true")
+      sql("select CAST(CAST('9999-12-31T12:13:14.56789Z' AS TIMESTAMP) AS INT)").collect()
```

Comment:
Can you use the timestamp literal instead of cast: `timestamp'9999-12-31T12:13:14.56789Z'`?